src/modules/sensors/sensors.cpp

```diff
@@ -325,41 +325,35 @@ int Sensors::parameters_update()
 		uint32_t device_id_mag   = calibration::GetCalibrationParamInt32("MAG",  "ID", i);
 
 		if (device_id_accel != 0) {
-			bool external_accel = (calibration::GetCalibrationParamInt32("ACC", "ROT", i) >= 0);
-			calibration::Accelerometer accel_cal(device_id_accel, external_accel);
+			calibration::Accelerometer accel_cal(device_id_accel);
 		}
 
 		if (device_id_gyro != 0) {
-			bool external_gyro = (calibration::GetCalibrationParamInt32("GYRO", "ROT", i) >= 0);
-			calibration::Gyroscope gyro_cal(device_id_gyro, external_gyro);
+			calibration::Gyroscope gyro_cal(device_id_gyro);
 		}
 
 		if (device_id_mag != 0) {
-			bool external_mag = (calibration::GetCalibrationParamInt32("MAG", "ROT", i) >= 0);
-			calibration::Magnetometer mag_cal(device_id_mag, external_mag);
+			calibration::Magnetometer mag_cal(device_id_mag);
 		}
 	}
 
 	// ensure calibration slots are active for the number of sensors currently available
 	// this to done to eliminate differences in the active set of parameters before and after sensor calibration
 	for (int i = 0; i < MAX_SENSOR_COUNT; i++) {
 		if (orb_exists(ORB_ID(sensor_accel), i) == PX4_OK) {
-			bool external = (calibration::GetCalibrationParamInt32("ACC", "ROT", i) >= 0);
-			calibration::Accelerometer cal{0, external};
+			calibration::Accelerometer cal{0};
 			cal.set_calibration_index(i);
 			cal.ParametersUpdate();
 		}
 
 		if (orb_exists(ORB_ID(sensor_gyro), i) == PX4_OK) {
-			bool external = (calibration::GetCalibrationParamInt32("GYRO", "ROT", i) >= 0);
-			calibration::Gyroscope cal{0, external};
+			calibration::Gyroscope cal{0};
 			cal.set_calibration_index(i);
 			cal.ParametersUpdate();
 		}
 
 		if (orb_exists(ORB_ID(sensor_mag), i) == PX4_OK) {
-			bool external = (calibration::GetCalibrationParamInt32("MAG", "ROT", i) >= 0);
-			calibration::Magnetometer cal{0, external};
+			calibration::Magnetometer cal{0};
```

Comment:
Double check this.

Reply:
Needed so that a system that boots up with an external mag disconnected doesn't lose it's calibrated when you make an unrelated parameter change.

Reply:
Fixed.